flag_engine/identities/models.py

```diff
@@ -32,22 +50,54 @@ class IdentityModel:
     def composite_key(self):
         return self.generate_composite_key(self.environment_api_key, self.identifier)
 
-    def get_all_feature_states(
-        self, environment: EnvironmentModel
-    ) -> typing.List[FeatureStateModel]:
-        all_feature_states = {fs.feature: fs for fs in environment.feature_states}
+    def _get_env_feature_states_dict(
+        self,
+        environment: EnvironmentModel,
+        feature_name: str = None,
+    ) -> typing.Dict[FeatureModel, FeatureStateModel]:
+        feature_states = environment.feature_states
+        if feature_name:
+            feature_states = [environment.get_feature_state(feature_name)]
+        return {fs.feature: fs for fs in feature_states}
+
+    def _update_feature_state_dict_with_segment_override(
+        self,
+        environment,
+        all_feature_states: typing.Dict[FeatureModel, FeatureStateModel],
+    ) -> None:
+        for feature_state in environment.segment_overrides:
+            segment = environment.get_segment(feature_state.segment_id)
+            if feature_state.feature in all_feature_states and self.in_segment(segment):
+                all_feature_states[feature_state.feature] = feature_state
 
-        for segment_override in environment.segment_overrides:
-            feature_state = segment_override.feature_state
-            feature = feature_state.feature
-            if self.in_segment(segment_override.segment):
-                all_feature_states[feature] = feature_state
+    def _update_feature_state_dict_with_identity_override(
```

Comment:
I'd rather keep this in the `get_all_feature_states` method to avoid having methods which mutate the input values. I find that having methods mutate the inputs causing a lot of confusion and makes testing difficult. Although I appreciate in this case that the methods are 'private'.